onnxruntime/core/optimizer/transpose_optimization/ort_transpose_optimization.cc

```diff
@@ -89,6 +89,28 @@ constexpr HandlerInfo max_pool_op_handler = {&FirstInput, &HandleMaxPool};
 constexpr HandlerInfo node_1_inp_handler = {&FirstInput, &HandleSimpleNode};
 constexpr HandlerInfo reduce_op_handler = {&FirstInput, &HandleReduceOps};
 
+static bool HandleContribQuantizeDequantizeLinear(HandlerArgs& args) {
+  if (!TransposeQuantizeDequantizeAxis(args.ctx.graph, args.perm, args.node)) {
+    return false;
+  }
+
+  TransposeFirstInput(args.ctx, args.node, args.perm_inv);
+  TransposeOutputs(args.ctx, args.node, args.perm);
+
+  return true;
+}
+
+constexpr HandlerInfo contrib_quantize_dequantize_linear_handler = {&FirstInput, &HandleContribQuantizeDequantizeLinear};
+
+const HandlerMap& OrtHandlers() {
+  static const HandlerMap extended_handler_map{
+      {"com.microsoft.QuantizeLinear", contrib_quantize_dequantize_linear_handler},
+      {"com.microsoft.DequantizeLinear", contrib_quantize_dequantize_linear_handler},
+  };
```

Comment:
Is there a scenario where we would want these but not the extended handlers to be enabled? 

The original intention was to separate the generic handling for onnx ops from any ORT specific things, so having multiple layers of ORT specific things wasn't expected. We can do it if needed - just wondering if that complexity is required.

Reply:
> Is there a scenario where we would want these but not the extended handlers to be enabled?

From my understanding, that is already the case. The `OrtExtendedHandlers()` are only used by the NhwcTransformer (level 3 optimization).

https://github.com/microsoft/onnxruntime/blob/6db72165eb3269e768f0fea28a7b210f588c7abe/onnxruntime/core/optimizer/nhwc_transformer.cc#L231

On the other hand, these `OrtHandlers()` are used (without the extended handlers) by the L1 transpose optimization and the layout transformer (for nhwc eps).

> The original intention was to separate the generic handling for onnx ops from any ORT specific things, so having multiple layers of ORT specific things wasn't expected. We can do it if needed - just wondering if that complexity is required.

Yeah, having multiple layers of ORT handlers is a bit confusing. It would be nice to consolidate. Before looking at your draft Resize PR, my first idea was to include the com.microsoft.QDQ ops in the onnx handlers directly. This gets rid of the multiple ort handlers, but the downside is that we would be mixing ort-specific handlers with onnx handlers.

Reply:
I'd prefer to keep the ONNX handlers clean.

Does it matter if the extended handlers are used in the L1 transpose optimization? They are either EP specific (need node to be assigned, which hasn't happened when L1 runs) or involve contrib ops (unless a pre-optimized model was EP specific ops was saved, the model wouldn't have any of those).

Reply:
I don't think it matters, but I am not sure. The MaxPool extended handler is only run if assigned to CPU EP, so that one is fine. However, the contrib op extended handlers don't check for EP. I'll try running local unit tests with code that always uses the extend handlers in L1 transpose opt **and** during layout transformation for nhwc eps.

Reply:
@skottmckay I added QDQ contrib ops to the extended handlers and am now passing in `OrtExtendedHandlers()` for L1 transpose optimization and during nhwc layout transformation. I had to ensure that the extended MaxPool handler returns false if contrib ops are disabled.

From my inspection of this area of the code, this change should be fine. However, is there someone who we should ask regarding use of the extended handlers in all calls to the transpose optimizer?